### PR TITLE
Fix triple+ quote parsing on Unix

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

**Comment by danmosemsft** [comment]
@wtgodbe since this is not one of those "self evidently correct" changes can you please include an explanation of the problem, how it came about and why this is the right fix? If you include in a commit message it will be included in the final commit also.

**Comment by wtgodbe** [comment]
@danmosemsft yes, I'm planning on updating the PR later on, right now I'm trying to get some test results from CI in parallel to doing stuff locally.

File: src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs [hunk 1]
```diff
@@ -553,16 +553,7 @@ private static string GetNextArgument(string arguments, ref int i)
...
-                    {
-                        currentArgument.Append('"');
-                        i++;
-                    }
-                    else
-                    {
-                        inQuotes = !inQuotes;
-                    }
-
+                    inQuotes = !inQuotes;
```
Comment (Priya91) [inline src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs]: requires adding more tests, and and updating the existing test cases, if none of the test cases broke with this change, then there was not sufficient test cases to begin with.

File: src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs [hunk 2]
```diff
@@ -265,6 +265,41 @@ public void TestStartOnUnixWithBadPermissions()
             Assert.NotEqual(0, e.NativeErrorCode);
         }
 
+        [Theory]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.UapNotUapAot, "Non applicable for uap - RemoteInvoke works differently")]
```
Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs]: Is this a duplication of the test in processtests.cs? Those tests are run on all platforms, why duplicate the test here again?

Comment (wtgodbe) [reply src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs]: Yes, I was under the impression those weren't running on Unix - I'm very surprised none of them failed in the PR before, then. For example, https://github.com/dotnet/corefx/pull/25866/files#diff-5bc7793999b3bb4572fb50d828365f4fR281 should definitely have failed with the previous state of this change
